### ISSUE TITLE
[P2] Range designator expansion cap (10,000) silently truncates without warning

### ISSUE THREAD

**Issue by ChaseWNorton** [anthropics/claudes-c-compiler#127]
## Problem

In PR #103 ([CC][Fix #84]), range designator expansion in `src/frontend/parser/declarations.rs` was capped at 10,000 elements to prevent hangs. However, the cap silently truncates ranges without emitting any diagnostic.

For example:
```c
int arr[100000] = {[0 ... 99999] = 1};
```

Only indices 0-9999 would be initialized to 1. The remaining 90,000 elements would silently be zero. No warning is emitted.

## Location

`src/frontend/parser/declarations.rs:~889`:
```rust
let count = (hi_val.saturating_sub(lo_val).saturating_add(1)).min(10_000);
for idx in lo_val..lo_val.saturating_add(count) {
```

## Expected behavior

When the range exceeds the expansion cap, the compiler should either:
1. Emit a warning: `range designator [0 ... 99999] truncated to 10000 elements` (preferred)
2. Or handle large ranges without expansion (emit a fill loop or memset-like pattern at the IR level)

## Acceptance criteria

- [ ] A warning is emitted when range designator expansion is truncated
- [ ] Or the cap is raised / removed in favor of a smarter IR-level approach
- [ ] `cargo test --lib` passes

Found during review of PR #103.
